timescaledb/how-to-guides/backup-and-restore/pg-dump-and-restore.md

```diff
@@ -1,114 +1,103 @@
- version installed.  (In particular, the backup could be for some
- version 1.6, but then the `CREATE EXTENSION timescaledb` command just
- installs the latest (say, 1.7), and thus does not have the
- opportunity to run our upgrade scripts.)  
-
-The workaround is that when restoring from a backup, you need to
- restore to a PostgreSQL instance with the same extension version
- installed, and *then* upgrade the version.
- </highlight>
-
 <highlight type="warning">
- These instructions do not work if you use flags to selectively
- choose tables (`-t`) or schemas (`--schema`), and so cannot be used
- to backup only an individual hypertable.  In particular, even if you
- explicitly specify both the hypertable and all of its constituent
- chunks, this dump would still lack necessary information that
- TimescaleDB stores in the database catalog about the relation between
- these tables.
-
-You can, however, explicitly *exclude* tables from this whole
- database backup (`-T`), as well as continue to selectively backup
- plain tables (i.e., non-hypertable) as well.
-</highlight>
-
-
-## Individual hypertables
-
-Below is the procedure for performing a backup and restore of hypertable `conditions`.
-
-### Backup
-
-Backup the hypertable schema:
-```bash
-pg_dump -s -d old_db --table conditions -N _timescaledb_internal | \
-  grep -v _timescaledb_internal > schema.sql
-```
-
-Backup the hypertable data to a CSV file.
-```bash
-psql -d old_db \
--c "\COPY (SELECT * FROM conditions) TO data.csv DELIMITER ',' CSV"
-```
-
-### Restore
-Restore the schema:
-```bash
-psql -d new_db < schema.sql
-```
-
-Recreate the hypertables:
-```bash
-psql -d new_db -c "SELECT create_hypertable('conditions', 'time')"
-```
-
-<highlight type="tip">
-The parameters to `create_hypertable` do not need to be
-the same as in the old db, so this is a good way to re-organize
-your hypertables (e.g., change partitioning key, number of
-partitions, chunk interval sizes, etc.).
+Do not use the `pg_dump` command to backup individual hypertables. Dumps created using this method lack the necessary information to correctly restore the hypertable from backup.
 </highlight>
 
-Restore the data:
-```bash
-psql -d new_db -c "\COPY conditions FROM data.csv CSV"
-```
+## Restore your entire database from backup
+When you need to restore data from a backup, you can use `psql` to create a new database and restore the data.
+
+### Procedure: Restoring an entire database from backup
+1.  In `pgsql`, create a new database to restore to, and connect to it:
+    ```sql
+    CREATE DATABASE exampledb;
+    \c exampledb
+    CREATE EXTENSION IF NOT EXISTS timescaledb;
+    SELECT timescaledb_pre_restore();
+    ```
+1.  Restore the database:
+    ```sql
+    \! pg_restore -Fc -d exampledb exampledb.bak
+    SELECT timescaledb_post_restore();
+    ```
+
+### Troubleshoot version mismatches
+
+The PostgreSQL `pg_dump` command does not allow you to specify which version of
```

Comment:
These two paragraphs needs some work to reduce duplication and redundancy. It does feel, to me, that these warnings are important enough that we should address them upfront. I mean, in theory folks aren't doing a backup and then ignoring the rest of the article... but people are lazy and it would be easy to miss this important point in my opinion.

At some point we also need to document David's backup tool, which has been on my plate for a long time. ☹️ 

https://github.com/timescale/timescaledb-backup

Reply:
> These two paragraphs needs some work to reduce duplication and redundancy. It does feel, to me, that these warnings are important enough that we should address them upfront. I mean, in theory folks aren't doing a backup and then ignoring the rest of the article... but people are lazy and it would be easy to miss this important point in my opinion.

I can see the point, but surely you would only need this info if you ran into an error with the backup/restore? In which case, you'd come looking for troubleshooting info?

> 
> At some point we also need to document David's backup tool, which has been on my plate for a long time. ☹️
> 
> https://github.com/timescale/timescaledb-backup

New issue: https://github.com/timescale/docs/issues/97